### PR TITLE
github issues completely rebuild if an update is found

### PR DESCRIPTION
**Pull request by m3brown**

### CODE REVIEW

File: server/indexers/gh_issues.py [hunk 1]
```diff
@@ -13,14 +13,25 @@ def index(gh_type, client, repo_name):
...
+    else:
+        print '%s (%s): skipping %s' % (repo_name, gh_type, obj_type)
     end = time.mktime(datetime.now().timetuple())
     print '%s: %s github issues (%s secs)' % (repo_name, len(bulk_data)/2, end-start)
 
 
-def index_gh_issues(gh_type, client, repo_name, since):
+def is_updated_issues(client, repo_name, since):
+    (owner, repo) = repo_name.split('/')
+    issues = client.repos._(owner)._(repo).issues.params(
```
Comment (dgreisen-cfpb) [inline server/indexers/gh_issues.py]: you can just do 

    issues = client.repos._(repo_name).issues.params(

no need to split it just to join it again

File: server/indexers/gh_issues.py [hunk 2]
```diff
@@ -13,14 +13,25 @@ def index(gh_type, client, repo_name):
...
+        state='all', since=since)
+    return True if issues.get().json() else False
```
Comment (dgreisen-cfpb) [inline server/indexers/gh_issues.py]: deserializing json is a really slow way to see if it return a 304.

    return issues.get().status_code != 304

File: server/indexers/gh_issues.py [hunk 3]
```diff
@@ -13,14 +13,25 @@ def index(gh_type, client, repo_name):
...
-    bulk_data = index_gh_issues(gh_type, client, repo_name, indexed_timestamp)
-    bulk_data += index_gh_issue_comments(gh_type, client, repo_name, indexed_timestamp)
-    helpers.update_repo_index(gh_type, repo_name, obj_type, bulk_data)
+    bulk_data = []
+    if is_updated_issues(client, repo_name, indexed_timestamp):
+        bulk_data += index_gh_issues(gh_type, client, repo_name)
+        bulk_data += index_gh_issue_comments(gh_type, client, repo_name)
+        helpers.rebuild_repo_index(gh_type, repo_name, obj_type, bulk_data)
+    else:
+        print '%s (%s): skipping %s' % (repo_name, gh_type, obj_type)
```
Comment (dgreisen-cfpb) [inline server/indexers/gh_issues.py]: will this print skipping and then print the line at 24? I don't think that's what we want

Comment (m3brown) [inline server/indexers/gh_issues.py]: Good call, I was trying to make the console output more consistent with the other data types, but this is redundant.

File: server/indexers/gh_issues.py (hunk 2 above)
Comment (m3brown) [inline server/indexers/gh_issues.py]: Unfortunately it returns `[]` with status code 200...

File: server/indexers/gh_issues.py (hunk 1 above)
Comment (m3brown) [inline server/indexers/gh_issues.py]: I swear I tried this last night, but tried it again just now and it worked.  Good call!

**Comment by dgreisen-cfpb** [comment]
looks good.

**Pull request merged**